legacy/firmware/ethereum.c

```diff
@@ -37,14 +37,17 @@
 #include "transaction.h"
 #include "util.h"
 
-/* maximum supported chain id.  v must fit in an uint32_t. */
-#define MAX_CHAIN_ID 2147483629
+/* maximum supported chain id.  v must fit in an uint64_t.
+ * (2^64 - 35) / 2
+ */
+#define MAX_CHAIN_ID 9223372036854775790
```

Comment:
this still seems invalid, the maximum value of `uint64_t` is `2^64 - 1`
so it should be one less.

Also it would be nicer if this was a calculation, something like:
```
#define MAX_CHAIN_ID ((ULLONG_MAX - 35) >> 1)
```

Reply:
> this still seems invalid

the value is OK but the comment is wrong

Reply:
Done